core/domain/stats_domain.py

```diff
@@ -63,13 +66,120 @@
+    time_spent_in_sec: float
+    rule_spec_str: Optional[str]
+    answer_str: Optional[str]
+
+
+class ExplorationIssueDict(TypedDict):
+    """Dictionary representing the ExplorationIssue object."""
+
+    issue_type: str
+    issue_customization_args: Dict[str, Dict[str, Any]]
+    playthrough_ids: List[str]
+    schema_version: int
+    is_valid: bool
+
+
+class PlaythroughDict(TypedDict):
+    """Dictionary representing the PlayThrough object."""
+
+    exp_id: str
+    exp_version: int
+    issue_type: str
+    issue_customization_args: Dict[str, Any]
+    actions: List[LearnerActionDict]
+
+
+class ExplorationIssuesDict(TypedDict):
+    """Dictionary representing the ExplorationIssues object."""
+
+    exp_id: str
+    exp_version: int
+    unresolved_issues: List[ExplorationIssueDict]
+
+
+class LearnerAnswerDetailsDict(TypedDict):
+    """Dictionary representing the LearnerAnswerDetail object."""
+
+    state_reference: str
+    entity_type: str
+    interaction_id: str
+    learner_answer_info_list: List[LearnerAnswerInfoDict]
+    accumulated_answer_info_json_size_bytes: int
+    learner_answer_info_schema_version: int
+
+
+class ExplorationStatsDict(TypedDict):
+    """Dictionary representing the ExplorationStats object."""
+
+    exp_id: str
+    exp_version: int
+    num_starts_v1: int
+    num_starts_v2: int
+    num_actual_starts_v1: int
+    num_actual_starts_v2: int
+    num_completions_v1: int
+    num_completions_v2: int
+    state_stats_mapping: Dict[str, Dict[str, int]]
+
+
+class ExplorationStatsFrontendDict(TypedDict):
+    """Dictionary representing the ExplorationStats object
+    for use in frontend."""
+
+    exp_id: str
+    exp_version: int
+    num_starts: int
+    num_actual_starts: int
+    num_completions: int
+    state_stats_mapping: Dict[str, Dict[str, int]]
+
+
+class LearnerActionDict(TypedDict):
+    """Dictionary representing the LearnerAction object."""
+
+    action_type: str
+    action_customization_args: Dict[str, Any]
+    schema_version: int
+
+
+class AnswerOccurrenceDict(TypedDict):
+    """Dictionary representing the AnswerOccurrence object."""
+
+    answer: Any
+    frequency: int
+
+
+class LearnerAnswerInfoDict(TypedDict):
+    """Dictionary representing LearnerAnswerInfo object."""
+
+    id: str
+    answer: Any
+    answer_details: str
+    created_on: str
+
+
 class ExplorationStats:
     """Domain object representing analytics data for an exploration."""
 
     def __init__(
-            self, exp_id, exp_version, num_starts_v1, num_starts_v2,
-            num_actual_starts_v1, num_actual_starts_v2, num_completions_v1,
-            num_completions_v2, state_stats_mapping):
+            self, exp_id: str, exp_version: int, num_starts_v1: int,
+            num_starts_v2: int, num_actual_starts_v1: int,
+            num_actual_starts_v2: int, num_completions_v1: int,
+            num_completions_v2: int, state_stats_mapping:
+            Dict[str, StateStats]) -> None:
```

Comment:
```suggestion
        self, 
        exp_id: str,
        exp_version: int, 
        num_starts_v1: int,
        num_starts_v2: int, 
        num_actual_starts_v1: int,
        num_actual_starts_v2: int, 
        num_completions_v1: int,
        num_completions_v2: int, 
        state_stats_mapping: Dict[str, StateStats]
    ) -> None:
```

make sure to fix this in other places too

Reply:
Done